android/src/main/java/com/reactnativestripesdk/pushprovisioning/PushProvisioningProxy.kt

```diff
@@ -0,0 +1,109 @@
+package com.reactnativestripesdk
+
+import android.app.Activity
+import android.app.Activity.RESULT_OK
+import android.content.Intent
+import android.util.Log
+import com.facebook.react.bridge.*
+import com.reactnativestripesdk.pushprovisioning.AddToWalletButtonView
+import com.reactnativestripesdk.pushprovisioning.EphemeralKeyProvider
+import com.reactnativestripesdk.pushprovisioning.TapAndPayProxy
+import com.stripe.android.pushProvisioning.PushProvisioningActivity
+import com.stripe.android.pushProvisioning.PushProvisioningActivityStarter
+
+
+object PushProvisioningProxy {
+  private const val TAG = "StripePushProvisioning"
+  private var description = "Added by Stripe"
+  private var tokenRequiringTokenization: ReadableMap? = null
+
+  fun getApiVersion(): String {
+    return try {
+      Class.forName("com.stripe.android.pushProvisioning.PushProvisioningActivity")
+      PushProvisioningActivity.API_VERSION
+    } catch (e: Exception) {
+      Log.e(TAG, "PushProvisioning dependency not found")
+      ""
+    }
+  }
+
+  fun invoke(
+          context: ReactApplicationContext,
+          view: AddToWalletButtonView,
+          cardDescription: String,
+          ephemeralKey: String,
+          token: ReadableMap?
+  ) {
+    try {
+      Class.forName("com.stripe.android.pushProvisioning.PushProvisioningActivityStarter")
+      description = cardDescription
+      tokenRequiringTokenization = token
+      createActivityEventListener(context, view)
+      DefaultPushProvisioningProxy().beginPushProvisioning(
+        context.currentActivity!!,
```

Comment:
is this safe?

Reply:
Since `currentActivity` is pretty much only ever `null` in React Native modules super early on during app startup, and this requires a touch event to trigger, I believe this is pretty safe. can add a null check though if you think it's necessary?

Reply:
I think it would be better to have a null check so we don't have unexpected crashes.

Reply:
agreed 👍  added here- https://github.com/stripe/stripe-react-native/pull/889/commits/187757831fd759dc13c09cf5327c86f67a4b76c4